pkg/gitconfig/config.go

```diff
@@ -355,6 +356,84 @@ func NewFromMap(data map[string]string) *Config {
 
 // LoadConfig tries to load a gitconfig from the given path.
 func LoadConfig(fn string) (*Config, error) {
+	c, err := loadConfig(fn)
+	if err != nil {
+		return nil, err
+	}
+	c.path = fn
+
+	loadedConfigs := map[string]struct{}{}
+	loadedConfigs[fn] = struct{}{}
+	configsToLoad := []string{}
+	includePaths, includeExists := c.GetAll("include.path")
+	if includeExists {
+		configsToLoad = append(configsToLoad, getPathsForNestedConfig(includePaths, c.path)...)
+	}
+	for len(configsToLoad) > 0 {
```

Comment:
In Go you'd return early, i.e. if there are no configs to load you should return and otherwise continue. This saves some nesting.